src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -268,7 +269,18 @@ export default class DefaultDeviceController
     await this.chooseAudioInputDevice(null);
     await this.chooseVideoInputDevice(null);
 
+    // Clean up the active audio device created when `chooseAudioInputDevice(null)' is called.
+    // The active video device has already been cleaned up when `chooseVideoInputDevice(null)` is called above.
+    let active = this.activeDevices['audio'];
+    this.releaseActiveDevice(active);
+    active = undefined;
+    delete this.activeDevices['audio'];
+
     // Tear down any Web Audio infrastructure we have hanging around.
+    DefaultDeviceController.audioBufferSourceNode.stop();
+    DefaultDeviceController.audioBufferSourceNode.disconnect();
+    DefaultDeviceController.audioBufferSourceNode.buffer = undefined;
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/AudioBufferSourceNode
>The AudioBufferSourceNode interface is an [AudioScheduledSourceNode](https://developer.mozilla.org/en-US/docs/Web/API/AudioScheduledSourceNode) which represents an audio source consisting of in-memory audio data, stored in an [AudioBuffer](https://developer.mozilla.org/en-US/docs/Web/API/AudioBuffer).

So the idea is that the `audioBufferSourceNode.buffer` stores the actual audio source data. And since the `AudioBufferSourceNode` is not garbage collected. This audio source data preserve in memory and cause the leak.

This line manually cleans up this audio source data, so it fix the memory leak issue.
```
DefaultDeviceController.audioBufferSourceNode.buffer = undefined;
```

Reply:
So here's the case:
Ideally if we can clean up the `audioBufferSourceNode` itself, the `audioBufferSourceNode.buffer` should be garbage collected as well. But somehow it preserve in the memory and I tried so many ways to clean it up but failed.

So I have to clean up the `audioBufferSourceNode.buffer` instead. Technically with only `    DefaultDeviceController.audioBufferSourceNode.buffer = undefined;` could fix this memory leak issue. 

But it's best practice to clean up the audio node like this:
```
 DefaultDeviceController.audioBufferSourceNode.stop();
 DefaultDeviceController.audioBufferSourceNode.disconnect();
```
So I keep these even they don't work now.

Reply:
> Did not see we do it for VF? Or we have issue with VF too?

For `emptyAudioInputDevice` it has no actual input so we need to create an `AudioBuffer` to serve as audio input source.

For VF, the audio input source are from the real devices. so it does not have `AudioBuffer`, thus no need to clean up.

Reply:
Nice. just want to check